benchexec/tools/ultimate.py

```diff
@@ -204,113 +204,116 @@ def _requires_ultimate_data(self, executable):
         # all versions before 0.1.24 do not require ultimatedata
         return not (int(major) == 0 and int(minor) < 2 and int(patch) < 24)
 
-    def cmdline(self, executable, options, tasks, propertyfile=None, rlimits=None):
-        if rlimits is None:
-            rlimits = {}
+    def cmdline(self, executable, options, task, resource_limits):
+        self._uses_propertyfile = task.property_file is not None
+        combined_options = options + [*task.options] if task.options else []
 
-        self._uses_propertyfile = propertyfile is not None
-        if _OPTION_NO_WRAPPER in options:
+        if _OPTION_NO_WRAPPER in combined_options:
             # do not use old wrapper script even if property file is given
             self._uses_propertyfile = False
-            propertyfile = None
-            options.remove(_OPTION_NO_WRAPPER)
+            combined_options.remove(_OPTION_NO_WRAPPER)
 
         if self._is_svcomp17_version(executable):
```

Comment:
Note that the new API explicitly guarantees that information about the tool version may be stored in a field of `self` and used by methods like `cmdline`, so if you wish, you could simplify this.